x-pack/metricbeat/module/openai/usage/usage_integration_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package usage
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	mbtest "github.com/elastic/beats/v7/metricbeat/mb/testing"
+)
+
+func TestFetch(t *testing.T) {
+	apiKey := time.Now().String() // to generate a unique API key everytime
+	usagePath := "/usage"
+	server := initServer(usagePath, apiKey)
+	defer server.Close()
+
+	f := mbtest.NewReportingMetricSetV2Error(t, getConfig(server.URL+"/usage", apiKey))
+
+	events, errs := mbtest.ReportingFetchV2Error(f)
+	if len(errs) > 0 {
+		t.Fatalf("Expected 0 error, has %d: %v", len(errs), errs)
+	}
```

Comment:
[Suggestion]
Use `require.Empty`. `require` will abort the test and you can also add the extra error message if you use `require.Emptyf`